ownCloud/Settings/AutoUploadSettingsSection.swift

```diff
@@ -0,0 +1,427 @@
+						self?.setupVideoAutoUpload(enabled: switchState)
+					})
+				}
+				}, title: "Auto Upload Videos".localized, value: self.userDefaults.instantUploadVideos, identifier: "auto-upload-videos")
+
+			photoBookmarkAndPathSelectionRow = StaticTableViewRow(subtitleRowWithAction: { [weak self] (_, _) in
+				self?.showAccountSelectionViewController(for: .photo)
+				}, title: "Photo upload path".localized, subtitle: "", accessoryType: .disclosureIndicator, identifier: AutoUploadSettingsSection.photoUploadBookmarkAndPathSelectionRowIdentifier)
+
+			videoBookmarkAndPathSelectionRow = StaticTableViewRow(subtitleRowWithAction: { [weak self] (_, _) in
+				self?.showAccountSelectionViewController(for: .video)
+				}, title: "Video upload path".localized, subtitle: "", accessoryType: .disclosureIndicator, identifier: AutoUploadSettingsSection.videoUploadBookmarkAndPathSelectionRowIdentifier)
+
+			self.add(row: instantUploadPhotosRow!)
+			self.add(row: instantUploadVideosRow!)
+
+			updateDynamicUI()
+		}
+	}
+
+	private func setupPhotoAutoUpload(enabled:Bool) {
+
+		if !enabled {
+			userDefaults.resetInstantPhotoUploadConfiguration()
+			postSettingsChangedNotification()
+			updateDynamicUI()
+		} else {
+			userDefaults.instantUploadPhotos = true
+			userDefaults.instantUploadPhotosAfter = Date()
+			if userDefaults.instantPhotoUploadPath == nil || userDefaults.instantPhotoUploadBookmarkUUID == nil {
+				showAccountSelectionViewController(for: .photo)
+			}
+		}
+	}
+
+	private func setupVideoAutoUpload(enabled:Bool) {
+		if !enabled {
+			userDefaults.resetInstantVideoUploadConfiguration()
+			postSettingsChangedNotification()
+			updateDynamicUI()
+		} else {
+			userDefaults.instantUploadVideos = true
+			userDefaults.instantUploadVideosAfter = Date()
+			if userDefaults.instantVideoUploadPath == nil || userDefaults.instantVideoUploadBookmarkUUID == nil {
+				showAccountSelectionViewController(for: .video)
+			}
+		}
+	}
+
+	private func getSelectedBookmark(for mediaType:MediaType) -> OCBookmark? {
+
+		var bookmarkUUID: UUID?
+
+		switch mediaType {
+		case .photo:
+			bookmarkUUID = self.userDefaults.instantPhotoUploadBookmarkUUID
+		case .video:
+			bookmarkUUID = self.userDefaults.instantVideoUploadBookmarkUUID
+		}
+
+		if let selectedBookmarkUUID = bookmarkUUID {
+			let bookmarks : [OCBookmark] = OCBookmarkManager.shared.bookmarks as [OCBookmark]
+			return bookmarks.filter({ $0.uuid == selectedBookmarkUUID}).first
+		}
+		return nil
+	}
+
+	private func updateDynamicUI() {
+
+		self.remove(rowWithIdentifier: AutoUploadSettingsSection.photoUploadBookmarkAndPathSelectionRowIdentifier)
+		self.remove(rowWithIdentifier: AutoUploadSettingsSection.videoUploadBookmarkAndPathSelectionRowIdentifier)
+
+		if let bookmark = getSelectedBookmark(for: .photo), let path = userDefaults.instantPhotoUploadPath {
+			OCItemTracker().item(for: bookmark, at: path) { (error, _, pathItem) in
```

Comment:
@felix-schwarz fixed `OCItemTracker`, discussed it with @hosy and both had an opinion that `OCItemTracker` should not leak `OCCore` instances since they are only used within `OCItemTracker` completion handler anyway.

Reply:
Thanks!